### ISSUE TITLE
BLOCK002 reports blocks preceded by a line comment and ignores the start of the file

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#64]
## Problem

BLOCK002 ("Block should be preceded by a blank line") has two gaps in its backward scan over block prefix lines:

1. Line comments (`// …`) directly before a block are treated as content, so a block preceded by a comment is reported.
2. The scan (`while prev_index > 0 and _is_block_prefix(...)`) stops at index 0 and never examines the first line itself, so the start of the file is not treated as a boundary.

Seen in a downstream docs project with a generated file that starts with a comment.

## Minimal reproduction

At commit `911440ac35d5849349389cce7ba04311d75dca49`, `file.adoc`:

```asciidoc
// generated file
[cols="1,1"]
|===
| a | b
|===
```

```
asciidoc-linter --format plain file.adoc
```

## Expected vs. actual

- Expected: no BLOCK002 finding; only a comment and an attribute line precede the table, at the start of the file.
- Actual: `Block should be preceded by a blank line`.

## Proposal

Treat `//` line comments like attribute and title lines when scanning backwards, and treat the start of the file as a valid boundary (include index 0 in the scan).

**Comment by rdmueller** [comment]
## Analysis

Confirmed at 2ded7ad. Cause: `BlockSpacingRule.check_line()` in `asciidoc_linter/rules/block_rules.py`:

```python
while prev_index > 0 and self._is_block_prefix(document[prev_index].strip()):
    prev_index -= 1
prev_line = document[prev_index].strip()
```

1. `_is_block_prefix()` knows only attribute lines (`[...]`) and block titles (`.Title`), not line comments (`// …`).
2. The loop stops at index 0 and then treats line 0 as the "previous line", even when line 0 is itself a prefix. So the start of the file never counts as a boundary.

**Fix plan (fix order: 2 of 9, small and independent)**
- `_is_block_prefix()` also accepts `//` line comments (but not `////`, which is a comment block delimiter).
- Scan down to index 0. If every line before the block is a prefix, the file start is the boundary → no finding.
- Tests: comment + attribute at the file start, comment between paragraph and block (must still be reported), block on line 1.

---
_Generated by [Claude Code](https://claude.ai/code)_
